Evaluate floating condition when calculating beta when marine check is on

matthewhoffman opened the pull request:
Previously, the marine check was adjusting the ice load for floating and grounded ice, but it was not applying the flotation condition when calculating beta.  This commit adds the flotation evaluation to beta.

matthewhoffman committed:
Evaluate floating condition when calculating beta when marine check is on

Previously, the marine check was adjusting the ice load for floating and
grounded ice, but it was not applying the flotation condition when
calculating beta.  This commit adds the flotation evaluation to beta.

matthewhoffman commented:
# Testing

Here are results using the MALI output from ISMIP6-2300 exp03 evaluated at 2015 and 2100 with the marine check on.  In each mask, red=1, blue=0.  I'm showing both time levels because the evaluation of beta at the initial time occurs in a different part of the code than for subsequent times.

![image](https://github.com/user-attachments/assets/4100dc70-38a8-46b8-8444-f5fa95f0291e)

hollyhan reviewed (approved):
@matthewhoffman , thanks for carefully observing and fixing the bug! Approving the PR.

matthewhoffman committed:
Fixup: define icestar properly on init if checkmarine is false

This was a mistake in the previous commit that broke the beta
calculation when the marine check is off
